features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -599,7 +617,12 @@ nsapi_error_t mbed_lwip_bringup_2(bool dhcp, bool ppp, const char *ip, const cha
 
     netif_set_default(&lwip_netif);
     netif_set_link_callback(&lwip_netif, mbed_lwip_netif_link_irq);
-    netif_set_status_callback(&lwip_netif, mbed_lwip_netif_status_irq);
+    if (status_cb) {
+        netif_set_status_callback(&lwip_netif, status_cb);
```

Comment:
Don't you always need to call our callback to unblock the blocking connect? Oh, you've assuming the person doing the attach will call back in to our status callback.

No that doesn't make sense, and will cause complication with the stack generalisation (#5202). The callback to lwIP should always be mbed_lwip_netif_status_irq. That can call status_cb if set - you'll need to record its setting.